File: problem-three.js
// This right here verifies that a number is prime or not prime!
function verifyPrime(number) {

	for(var i = 2; i < number; i++) {

  		if(number % i === 0) {

    		return false;

 		 } 

	}

	return true;
	
}

function largestPrimeFactor(number) {

	var factor;

	for(var i = 0; i <= number; i++) {

		if( Number.isInteger(number / i) ) {

			if ( verifyPrime(i) ) {

				number = number / i;

				factor = i;

			}

		}

	}

	return factor;

}

console.log(largestPrimeFactor(600851475143));

File: problem-three.rb
# duly note: I am not strong on blocks, procs, and methods that call them. Need more practice


require 'prime'

def largestPrimeFactor(number)

	factor = nil

	for i in 1..number

		if( (number / i).is_a? Integer ) 

			if Prime.prime?(i) 

				number = number / i

				factor = i

			end

		end

	end

	return factor;

end

puts largestPrimeFactor(600851475143)

# still unsolved! Hitting an infinite loop somewhere.